src/commands/startStreamingLogs.ts

```diff
@@ -5,27 +5,40 @@
 
 import * as vscode from 'vscode';
 import * as appservice from 'vscode-azureappservice';
+import { LogStreamTreeItem } from '../explorer/LogStreamTreeItem';
 import { SiteTreeItem } from "../explorer/SiteTreeItem";
 import { WebAppTreeItem } from '../explorer/WebAppTreeItem';
 import { ext } from '../extensionVariables';
 import { enableFileLogging } from './enableFileLogging';
 
-export async function startStreamingLogs(node?: SiteTreeItem): Promise<void> {
+export async function startStreamingLogs(node?: SiteTreeItem | LogStreamTreeItem): Promise<void> {
+    let logStreamTreeItem: LogStreamTreeItem | undefined;
+    let webAppTreeItem: WebAppTreeItem;
+
     if (!node) {
-        node = <WebAppTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+        webAppTreeItem = <WebAppTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    } else if (node.contextValue === LogStreamTreeItem.contextValue) {
+        logStreamTreeItem = <LogStreamTreeItem>node;
+        // tslint:disable-next-line:no-non-null-assertion
+        webAppTreeItem = <WebAppTreeItem>node.parent!.parent;
+    } else {
+        webAppTreeItem = <WebAppTreeItem>node;
     }
 
     const verifyLoggingEnabled: () => Promise<void> = async (): Promise<void> => {
         const isEnabled = await vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async p => {
             p.report({ message: 'Checking container diagnostics settings...' });
-            // tslint:disable-next-line:no-non-null-assertion
-            return await node!.isHttpLogsEnabled();
+            return await webAppTreeItem.isHttpLogsEnabled();
         });
         if (!isEnabled) {
-            // tslint:disable-next-line:no-non-null-assertion
-            await enableFileLogging(node!);
+            await enableFileLogging(webAppTreeItem);
         }
     };
 
-    await appservice.startStreamingLogs(node.root.client, verifyLoggingEnabled, node.logStreamLabel);
+    await appservice.startStreamingLogs(webAppTreeItem.root.client, verifyLoggingEnabled, webAppTreeItem.logStreamLabel);
```

Comment:
startStreamingLogs already returns the `ILogStream`. I'm pretty sure you can use that rather than searching for the log stream using your new `getLogStreams` thing